packages/lit-dev-content/samples/examples/react-basics/demo-greeting.ts

```diff
@@ -8,7 +8,32 @@ export class DemoGreeting extends LitElement {
   render() {
     return html`
       <p>Hello, ${this.name}!</p>
-      <p>Checkout our <a href="https://lit.dev/docs/frameworks/react/" target="_blank" rel="noopener noreferrer">React framework docs</a>!</p>
+      <p>
+        Checkout our
+        <a
+          href="https://lit.dev/docs/frameworks/react/"
+          target="_blank"
+          rel="noopener noreferrer"
+          >React framework docs</a
+        >!
+      </p>
     `;
   }
 }
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'demo-greeting': DemoGreeting;
+  }
+
+  namespace JSX {
```

Comment:
I don't think we should put this with the element file. We don't really expect element authors to add React JSX types too. This is one reason to use the wrapper even for simple elements - you get good types automatically.

If you want to show this I'd put it in app.tsx and try to make it handle all elements with a mapped type. I got something kind of working along these lines for Solid once: https://github.com/shoelace-style/shoelace/discussions/770#discussioncomment-2852125

I'm not 100% sure that didn't mess up types for built-in elements though.

Reply:
Thanks for the link! If we end up wanting to keep a bare custom element usage, I'll try this.